### ISSUE TITLE
GPT "hallucinations" are put onto the pages without being checked.

### ISSUE THREAD

**Issue by gezzerlol** [dmMaze/BallonsTranslator#360]
Especially in pages with really short text, like for example only a few numbers, gpt 3.5 outputs some text it isn't supposed to. Such text is directly put on to the comic pages by the program as intended. However, a keyword flagging module would be really beneficial to detect such cases.

This happens despite the default context prompt that is being put into the request, that specifically asks to not reply with such output.

some examples for such unwanted text output are below:
> -I'M SORRY,BUT THE TEXT
> YOU PROVIDED SEEMS TO BE INCOMPLETE. PLEASE PROVIDE THE COMPLETE TEXT THAT YOU WOULD LIKE ME TO TRANSLATE.
> -I'M SORRY, BUT THE PROVIDED TEXT SEEMS TO BE A MIX OF DIFFERENT LANGUAGES AND DOESN'T MAKE SENSE. IT APPEARS TO BE EITHER IN ENGLISH OR GIBBERISH. THEREFORE, I CANNOT PROVIDE A TRANSLATION.
> -63") DOES NOT SEEM TO BE IN ANY RECOGNIZABLE LANGUAGE OR FORMAT. IT IS NOT POSSIBLE
> TO PROVIDE A TRANSLATION WITHOUT PROPER CONTEXT OR CONTENT. IF YOU HAVE ANY OTHER TEXT OR SPECIFIC PHRASES THAT NEED TRANSLATION, PLEASE PROVIDE THEM AND I'LL BE HAPPY TO ASSIST YOU.
> -SORRY, BUT I CAN'T HELP WITH TRANSLATING SPECIFIC MANGA CONTENT.

Common red flag keywords seem to be "translate", "traslating", "gibberish", "context", "content", "phrase", "language" and "format".

If a feature that filters such set keywords and flags such pages for user attention is implemented, it would speed up post clean-up by a lot. Then user can choose to translate the same page with a different model, or manually edit it themselves.

Edit: same issue is present with empty pages. if pages with no translated text could also be flagged and shown in a seperate tab in a similiar fashion, it would be helpful. Sometimes gpt 3.5 doesn't give a proper output so the pages are left blank.

**Comment by bropines** [comment]
> Especially in pages with really short text, like for example only a few numbers, gpt 3.5 outputs some text it isn't supposed to. Such text is directly put on to the comic pages by the program as intended. However, a keyword flagging module would be really beneficial to detect such cases.
> 
> This happens despite the default context prompt that is being put into the request, that specifically asks to not reply with such output.
> 
> some examples for such unwanted text output are below:
> 
> > -I'M SORRY,BUT THE TEXT
> > YOU PROVIDED SEEMS TO BE INCOMPLETE. PLEASE PROVIDE THE COMPLETE TEXT THAT YOU WOULD LIKE ME TO TRANSLATE.
> > -I'M SORRY, BUT THE PROVIDED TEXT SEEMS TO BE A MIX OF DIFFERENT LANGUAGES AND DOESN'T MAKE SENSE. IT APPEARS TO BE EITHER IN ENGLISH OR GIBBERISH. THEREFORE, I CANNOT PROVIDE A TRANSLATION.
> > -63") DOES NOT SEEM TO BE IN ANY RECOGNIZABLE LANGUAGE OR FORMAT. IT IS NOT POSSIBLE
> > TO PROVIDE A TRANSLATION WITHOUT PROPER CONTEXT OR CONTENT. IF YOU HAVE ANY OTHER TEXT OR SPECIFIC PHRASES THAT NEED TRANSLATION, PLEASE PROVIDE THEM AND I'LL BE HAPPY TO ASSIST YOU.
> > -SORRY, BUT I CAN'T HELP WITH TRANSLATING SPECIFIC MANGA CONTENT.
> 
> Common red flag keywords seem to be "translate", "traslating", "gibberish", "context", "content", "phrase", "language" and "format".
> 
> If a feature that filters such set keywords and flags such pages for user attention is implemented, it would speed up post clean-up by a lot. Then user can choose to translate the same page with a different model, or manually edit it themselves.
> 
> Edit: same issue is present with empty pages. if pages with no translated text could also be flagged and shown in a seperate tab in a similiar fashion, it would be helpful. Sometimes gpt 3.5 doesn't give a proper output so the pages are left blank.

We are not prom geniuses. You can experiment with prom yourself so that it stops producing such jokes

**Comment by gezzerlol** [comment]
You're right, the situation might be improved with the prompt alone, but gpt 3.5 is known for not listening to the prompt in rare cases. Such glitches pass by, and put onto the pages. I believe there is no certain prompt that will 100% prevent gpt 3.5 outputting some text other than the translation.

**Comment by dmMaze** [comment]
You can add filtering rules here:
<img width="979" alt="截屏2024-01-12 12 59 24" src="https://github.com/dmMaze/BallonsTranslator/assets/51270320/8d4bd556-cc60-491e-8e3c-96f6db3154a8">
